File: Go/water_bottles/1518.go
package waterbottles

func numWaterBottles(numBottles int, numExchange int) int {
	drank := 0
	for numBottles >= numExchange {
		rem := numBottles % numExchange
		drank += numBottles - rem
		numBottles = (numBottles / numExchange) + rem
	}
	return drank + numBottles
}

File: TypeScript/water_bottles/1518.ts
function numWaterBottles(numBottles: number, numExchange: number): number {
    let result = 0;
    while (numBottles >= numExchange) {
      const rest = numBottles % numExchange;
      result = result + numBottles - rest;
      numBottles = Math.floor(numBottles / numExchange) + rest;
    }
    return result + numBottles;
  }
  
  export default numWaterBottles;